tools/topology/topology2/include/components/copier.conf

```diff
@@ -147,6 +147,18 @@ Class.Widget."copier" {
 		token_ref	"sof_tkn_copier.word"
 	}
 
+	DefineAttribute."fast_mode" {
+		type	"string"
+		# Token set reference name
+		token_ref	"sof_tkn_copier.bool"
+		constraints {
+			!valid_values [
```

Comment:
The bool value is a subset of integer in C lang and topology, assign "true" or "false" string to it is not going to work. the below code will allow us to write something like `mix_type    mix_in`, but in topology generation, mix_type get the value 0.
```
	DefineAttribute."mix_type" {
		type "string"
                token_ref       "sof_tkn_mixinout.word"
                constraints {
                        !valid_values [
                                "mix_in"
                                "mix_out"
                        ]
                        !tuple_values [
                                0
                                1
                        ]
                }
	}
```

Reply:
@jsarha Are you sure the fast_mode can be assigned correctly with 
```
!valid_values [
    "true"
    "false"
]
```
instead of
```
!valid_values [
    "true"
    "false"
]
!tuple_values [
   1
   0
]
```

Reply:
The original is not the first instance of notation I used: https://github.com/thesofproject/sof/blob/1319163ea2c6d8bea6fd7189dea6275ac5e34bf5/tools/topology/topology2/include/dais/ssp.conf#L78

However, I think the constraint section is not really needed. There are other attributes declared without it.